CHANGES.md

```diff
@@ -1,6 +1,472 @@
+  - added a color-coded version selection menu to the online documentation
+
+File / Directory Structure
+--------------------------
+* made `make install` install mason
+* added the CHPL_COMM_SUBSTRATE setting to GASNet-specific paths
+
+New Tools / Tool Changes
+------------------------
+* added support for bash tab completion for `chpl` compiler options
+  (see https://chapel-lang.org/docs/1.17/usingchapel/compiling.html#tab-completion-for-flags)
+* changed the generated executable's name from the main module to its filename
+  (see the `-o` flag on https://chapel-lang.org/docs/1.17/usingchapel/man.html)
+* added support for multiple mason repositories, including local registries
+  (see https://chapel-lang.org/docs/1.17/tools/mason/mason.html#local-registries)
+* added support for `mason help` and `mason version` commands
+* `chpl` now offers suggestions for unrecognized flags
+  (see error message for `chpl -fast`)
+* rewrote `printchplenv` and reworked its interface and output
+  (see `util/printchplenv --help`)
+* rewrote `compileline`
+  (see `util/config/compileline --help`))
+
+Semantic Changes / Changes to Chapel Language
+---------------------------------------------
+* `in` intents now behave more like variable initialization
+  (see https://github.com/chapel-lang/chapel/blob/master/doc/rst/developer/chips/13.rst)
+* ranges are now passed by `const in` intent by default
+  (see 'The Default Intent' in the 'Procedures' chapter of the specification)
+* disallowed implicit coercions for type arguments with a formal type specifier
+  (see 'Implicit Conversions' in the specification's 'Conversions' chapter)
+* changed the alignment of a non-stridable range from 0 to its low bound
+  (i.e., `(3..5).alignment` is now 3 where it used to be 0)
+
+New Features
+------------
+* added support for uninterpreted triple-quoted strings
+  (see 'Literals' section in 'Lexical Structure' chapter of the specification)
+* added support for `continue` statements within forall loops
+* extended `delete` to accept arrays and list of expressions to delete
+  (see 'Deleting Class Objects' in the 'Classes' chapter of the specification)
+* added an `isEmpty()` method on domains
+* type and param fields can now be accessed from a type variable
+  (see 'Field Accesses' in the 'Classes' chapter of the specification)
+* added support for forwarding methods on arrays, domains, and distributions
+  (see https://chapel-lang.org/docs/1.17/technotes/dsi.html#overview)
+* added support for associative domains whose index types contain ranges
+* added support for subtype queries on distributions
+  (see 'Methods on All Domain Types' in the 'Domains' chapter of the spec)
+* added support for using GASNet's `smp` conduit
+
+Feature Improvements
+--------------------
+* dramatically improved support for user-defined initializers
+  (see https://chapel-lang.org/docs/1.17/technotes/initializers.html)
+* made `real(32)` variables significantly more usable
```

Comment:
@mppf: On this item, when I was scanning my own notes, I'd written this more as:

`* improved param coercions between int/real/complex to real/complex`

Did I read too much into that effort, or did you just describe it differently?

Reply:
It's true that it does that, but it also helps with some cases that aren't param, e.g. int(32) to real(32).

Reply:
Thanks for the additional detail.  How about

```
* improved `enum` and numeric coercions, greatly improving `real(32)` usability
```

Reply:
Great!